chart/kubeapps/templates/kubeapps-jobs-upgrade.yaml

```diff
@@ -0,0 +1,61 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ template "kubeapps.kubeapps-jobs-upgrade.fullname" . }}
+  annotations:
+    helm.sh/hook: post-upgrade
+    helm.sh/hook-delete-policy: hook-succeeded
+  labels:
+    app: {{ template "kubeapps.kubeapps-jobs-upgrade.fullname" . }}
+    chart: {{ template "kubeapps.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  template:
+    metadata:
+      labels:
+        app: {{ template "kubeapps.kubeapps-jobs-upgrade.fullname" . }}
+        release: {{ .Release.Name }}
+    spec:
+      containers:
+      - name: kubectl
+        image: {{ template "kubeapps.image" .Values.apprepository.jobsImage }}
+        command:
+        - /bin/sh
+        - -c
+        args:
+        - |-
+            # Recreate MongoDB secret if doesn't exists (happens for versions < 1.0.0-beta.1)
+            if kubectl get secrets -n {{ .Release.Namespace }} {{ template "kubeapps.mongodb-secret.fullname" . }}; then
+              # Secret exists, do nothing
+              echo "MongoDB secret found"
+            else
+              # Secret doesn't exists, create it and restart mongodb
+              kubectl create secret generic -n {{ .Release.Namespace }} {{ template "kubeapps.mongodb-secret.fullname" . }} --from-literal mongodb-root-password=`cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1`
+              kubectl delete pods -n {{ .Release.Namespace }} -l app=mongodb,release={{ .Release.Name }}
+            fi
+            # Re-sync repositories
+            kubectl apply -f /tmp/apprepositories/apprepositories.yaml
```

Comment:
I am a little bit worried that this will make this issue even worse https://github.com/kubeapps/kubeapps/issues/668. It might make sense to bump it to 1.0? cc/ @andresmgot @prydonius ?

Reply:
I'll move the repository sync just in case the mongodb pod is restarted.